x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/index.ts

```diff
@@ -492,6 +492,46 @@ export class KnowledgeBaseService {
     };
   };
 
+  getInstructions = async (
+    user: { name: string },
+    namespace: string
+  ): Promise<UserInstruction[]> => {
+    try {
+      const response = await this.dependencies.esClient.search<KnowledgeBaseEntry>({
+        index: this.dependencies.resources.aliases.kb,
+        query: {
+          bool: {
+            must: [
+              {
+                term: {
+                  'labels.category.keyword': {
+                    value: 'instruction',
+                  },
+                },
+              },
+            ],
+            filter: getAccessQuery({
+              user,
+              namespace,
+            }),
+          },
+        },
+        size: 500,
+        _source: false,
+        fields: ['doc_id', 'text'],
```

Comment:
can we use `_source` here, `fields` can be much slower - mostly when using many fields but generally I think it's good to use `_source` unless there is a compelling reason to use `fields`

Reply:
Fixed in https://github.com/elastic/kibana/pull/180263/commits/65f34228d70728c795ff145a520001c48812acd0